eppo_client/client.py

```diff
@@ -36,143 +34,145 @@ def __init__(
-            raise e
-
-    def get_json_string_assignment(
-        self, subject_key: str, flag_key: str, subject_attributes=dict()
-    ) -> Optional[str]:
-        try:
-            assigned_variation = self.get_assignment_variation(
-                subject_key, flag_key, subject_attributes, VariationType.JSON
-            )
-            return (
-                assigned_variation.value
-                if assigned_variation is not None
-                else assigned_variation
-            )
-        except ValueError as e:
-            # allow ValueError to bubble up as it is a validation error
-            raise e
-        except Exception as e:
-            if self.__is_graceful_mode:
-                logger.error("[Eppo SDK] Error getting assignment: " + str(e))
-                return None
-            raise e
+        variation_jsons = self.get_assignment_variation(
+            subject_key,
+            flag_key,
+            subject_attributes,
+            VariationType.JSON,
+            default=default,
+        )
+        if variation_jsons:
+            return json.loads(variation_jsons)
 
     @deprecated(
         "get_assignment is deprecated in favor of the typed get_<type>_assignment methods"
     )
     def get_assignment(
-        self, subject_key: str, flag_key: str, subject_attributes=dict()
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[Dict[str, Union[str, float, int, bool]]] = None,
+        default=None,
     ) -> Optional[str]:
+        return self.get_assignment_variation(
+            subject_key, flag_key, subject_attributes, default=default
+        )
+
+    def get_assignment_variation(
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[Dict[str, Union[str, float, int, bool]]] = None,
+        expected_variation_type: Optional[VariationType] = None,
+        default=None,
+    ):
         try:
-            assigned_variation = self.get_assignment_variation(
-                subject_key, flag_key, subject_attributes
-            )
-            return (
-                assigned_variation.value
-                if assigned_variation is not None
-                else assigned_variation
+            result = self.get_assignment_detail(
+                subject_key, flag_key, subject_attributes, expected_variation_type
             )
+            if not result or not result.variation:
+                return default
+            return result.variation.value
         except ValueError as e:
             # allow ValueError to bubble up as it is a validation error
             raise e
         except Exception as e:
             if self.__is_graceful_mode:
                 logger.error("[Eppo SDK] Error getting assignment: " + str(e))
-                return None
+                return default
             raise e
 
-    def get_assignment_variation(
+    def get_assignment_detail(
```

Comment:
@aarsilv this function returns the FlagEvaluation object, which I currently consider to be the "detailed view of assignment".

Reply:
Is this part of the "public" API then? Developers who want the full return object call this one instead of get_<type>_assignment?

Reply:
Yes that is my thought -- but open to feedback. 

For example: should we have typed versions of the detailed version?